src/mds/Server.cc

```diff
@@ -1133,6 +1134,11 @@ void Server::handle_conf_change(const std::set<std::string>& changed) {
   if (changed.count("mds_recall_max_decay_rate")) {
     recall_throttle = DecayCounter(g_conf().get_val<double>("mds_recall_max_decay_rate"));
   }
+  if (changed.count("mds_max_snaps_per_dir")) {
```

Comment:
You also need to add this config option to `MDSRankDispatcher::get_tracked_conf_keys()`

Reply:
> You also need to add this config option to `MDSRankDispatcher::get_tracked_conf_keys()`

This helped. Thanks.